components/x-gift-article/src/GiftArticle.jsx

```diff
@@ -150,27 +150,38 @@ const withGiftFormActions = withActions(
 
 			activate() {
 				return async (state) => {
+					const { enabled, limit, hasCredits, firstTimeUser, requestAccess } =
+						await enterpriseApi.getEnterpriseArticleAllowance()
+
+					if (enabled) {
+						tracking.initEnterpriseSharing(
+							requestAccess
+								? 'enterprise-request-access'
+								: !hasCredits
+								? 'enterprise-no-credits'
+								: 'enterprise-enabled'
+						)
+					} else {
+						tracking.initEnterpriseSharing('enterprise-disabled')
+					}
+
 					if (initialProps.isFreeArticle) {
 						const { url, isShortened } = await api.getShorterUrl(state.urls.nonGift)
 
 						if (isShortened) {
-							return updaters.setShortenedNonGiftUrl(url)(state)
+							updaters.setShortenedNonGiftUrl(url)(state)
+						}
+						return {
+							invalidResponseFromApi: true,
+							enterpriseEnabled: enabled,
+							enterpriseLimit: limit,
+							enterpriseHasCredits: hasCredits,
+							enterpriseFirstTimeUser: firstTimeUser,
+							enterpriseRequestAccess: requestAccess
```

Comment:
Thought:
This seems to be the same return we have [here](https://github.com/Financial-Times/x-dash/pull/684/files#diff-7082a9e3cb317538577308da954ff347e6c58e103516faaa15ec7c405cba1161R198). If they need to stay in sync it is probably better to have a CONST object to return in both cases. You can name it as a significant state.

Reply:
I'm not quite sure I follow - which bit is the repeated return?

Reply:
What we are returning seems identical to the part [here](https://github.com/Financial-Times/x-dash/pull/684/files#diff-7082a9e3cb317538577308da954ff347e6c58e103516faaa15ec7c405cba1161R197)